src/desktop/apps/article/components/InfiniteScrollNewsArticle.js

```diff
@@ -0,0 +1,167 @@
+    marginTop: PropTypes.string,
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isLoading: false,
+      articles: [this.props.article],
+      offset: 0,
+      error: false,
+      following: setupFollows() || null,
+      isEnabled: true,
+    }
+  }
+
+  componentDidMount() {
+    setupFollowButtons(this.state.following)
+  }
+
+  fetchNextArticles = async () => {
+    const { articles, offset } = this.state
+
+    this.setState({
+      isLoading: true,
+    })
+
+    try {
+      const data = await positronql({
+        query: newsArticlesQuery({
+          offset,
+          limit: 3,
+          omit: this.props.article.id,
+        }),
+      })
+
+      const newArticles = data.articles
+
+      if (newArticles.length) {
+        this.setState({
+          articles: articles.concat(newArticles),
+          isLoading: false,
+          offset: offset + 3,
+        })
+        setupFollowButtons(this.state.following)
+      } else {
+        this.setState({
+          isEnabled: false,
+          isLoading: false,
+        })
+      }
+    } catch (error) {
+      console.error(
+        '(apps/article/InfiniteScrollNewsArticle) Error fetching next article set: ',
+        error
+      )
+
+      this.setState({
+        isEnabled: false,
+        isLoading: false,
+        error: true,
+      })
+    }
+  }
+
+  renderWaypoint = () => {
+    const { isEnabled, isLoading, error } = this.state
+
+    if (isEnabled) {
+      if (!isLoading) {
+        return (
+          <Waypoint
+            onEnter={this.fetchNextArticles}
+            threshold={2.0}
+            topOffset={FETCH_TOP_OFFSET}
+          />
+        )
+      } else if (!error) {
+        return (
+          <LoadingSpinner>
+            <div className="loading-spinner" />
+          </LoadingSpinner>
+        )
+      }
+    }
+  }
+
+  onEnter = (article, { previousPosition, currentPosition }) => {
+    if (previousPosition === 'above' && currentPosition === 'inside') {
+      document.title = article.thumbnail_title
+      window.history.replaceState({}, article.id, `/news/${article.slug}`)
+    }
+  }
+
+  onLeave = (i, { previousPosition, currentPosition }) => {
+    const nextArticle = this.state.articles[i + 1]
+
+    if (
+      nextArticle &&
+      previousPosition === 'inside' &&
+      currentPosition === 'above'
+    ) {
+      document.title = nextArticle.thumbnail_title
+      window.history.replaceState(
+        {},
+        nextArticle.id,
+        `/news/${nextArticle.slug}`
+      )
+    }
+  }
+
+  renderContent = () => {
+    return _.flatten(
+      this.state.articles.map((article, i) => {
+        return (
+          <div key={`article-${i}`}>
+            <Article
+              article={article}
+              relatedArticlesForCanvas={article.relatedArticlesCanvas}
+              isTruncated={i !== 0}
+              isMobile={this.props.isMobile}
+              display={article.display}
```

Comment:
Same here - we are injecting manually after RA Canvas. Which actually brings up a cool idea - what if we made [this number](https://github.com/artsy/force/pull/2282/files#r176595614) to 6 so the query can also return a fresh ad?